crates/node/builder/src/components/mod.rs

```diff
@@ -55,7 +55,7 @@ pub trait NodeComponents<T: FullNodeTypes>: Clone + Unpin + Send + Sync + 'stati
         + 'static;
 
     /// Network API.
-    type Network: FullNetwork<Client: BlockClient<Block = BlockTy<T::Types>>>;
+    type Network: FullNetwork<Client: BlockClient<Block = BlockTy<T::Types>>> + NetworkProtocols;
```

Comment:
I'd rather move this to `FullNetwork` directly because this is already an extension trait

Reply:
actually, it is already, so don't think this is actually required here?